go/ephemeral/device_ek_storage.go

```diff
@@ -197,6 +198,32 @@ func (s *DeviceEKStorage) GetAll(ctx context.Context) (deviceEKs DeviceEKMap, er
 	return s.getCache(ctx)
 }
 
+func (s *DeviceEKStorage) GetAllActive(ctx context.Context, now keybase1.Time) (metadatas []keybase1.DeviceEkMetadata, err error) {
+	defer s.G().CTrace(ctx, "DeviceEKStorage#GetAll", func() error { return err })()
+	s.Lock()
+	defer s.Unlock()
+
+	cache, err := s.getCache(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	activeKeysInOrder := []keybase1.DeviceEkMetadata{}
+	for _, deviceEK := range cache {
+		// Skip expired keys. Expired keys are spared from deletion past for a
+		// window past their expiry date, in case they're needed for
+		// decryption, but they're never signed over or used for encryption.
+		if now.UnixSeconds()-deviceEK.Metadata.Ctime.UnixSeconds() > KeyLifetimeSecs {
+			continue
+		}
+		// Collect out of order, then sort below.
+		activeKeysInOrder = append(activeKeysInOrder, deviceEK.Metadata)
+	}
+	sort.Slice(activeKeysInOrder, func(a, b int) bool { return activeKeysInOrder[a].Generation < activeKeysInOrder[b].Generation })
```

Comment:
why do we have to sort?

Reply:
It's part of the protocol that the existing list is in generation order. When the server checks those keys, it walks them in order and looks for equality.

Reply:
let's just put the sorting server side? it's really an implementation for it's check no?